src/network/networkctl.c

```diff
@@ -1611,6 +1615,25 @@ static int link_status_one(
                                    TABLE_UINT16, info->tunnel_port);
                 if (r < 0)
                         return table_log_add_error(r);
+
+        } else if (STRPTR_IN_SET(info->netdev_kind, "macvlan", "macvtap")) {
+                static const struct {
+                        const char *mode;
+                } mode_table[] = {
+                        [ MACVLAN_MODE_PRIVATE ]  = { "private" },
+                        [ MACVLAN_MODE_VEPA ]     = { "vepa" },
+                        [ MACVLAN_MODE_BRIDGE ]   = { "bridge" },
+                        [ MACVLAN_MODE_PASSTHRU ] = { "passthru" },
+                        [ MACVLAN_MODE_SOURCE ]   = { "source" },
+                };
+
+                r = table_add_many(table,
+                                   TABLE_EMPTY,
+                                   TABLE_STRING, "Mode:",
+                                   TABLE_STRING, mode_table[info->macvlan_mode]);
```

Comment:
Hmm, no, that is rather dangerous. We cannot trust the data send from the remote side blindly. Apart from bugs, the remote might get support for newer macvlan modes and then we'd just crash here. I'd very much prefer to have those tables in a separate file (to not clutter the networkctl.c file too much), and go through a proper accessor function.